feat(api):  add additional parameter to project/group iteration search (#2796)
python-gitlab/python-gitlab: docs/gl_objects/iterations.rst

List iterations for a group::

    iterations = group.iterations.list()

> Unavailable filters or filters conflicts::

Reviewer: ```suggestion
Unavailable filters or keyword conflicts::
```

>     In case you are trying to pass a parameter that is not supported or
>     collides with a python keyword (i.e. `in`) you have to use the
>     `query_parameters` argument:

Reviewer: There aren't really "unsupported" filters (since we just pass whatever the user provides to the API), only those that clash with other keywords. I'd maybe phrase it like this:
```suggestion
    In case you are trying to pass a parameter that collides with a python
    keyword (i.e. `in`) or with python-gitlab's internal arguments, you'll have
    to use the `query_parameters` argument:
```
